bluesky/suspenders.py

```diff
@@ -97,35 +99,42 @@ def _should_resume(self, value):
         """
         raise NotImplementedError()
 
-    def __call__(self, **kwargs):
+    def __call__(self, value, **kwargs):
         """
         Make the class callable so that we can
         pass it off to the pyepics callback stack.
 
         This expects the massive blob that comes from pyepics
         """
-        value = kwargs['value']
 
         with self._lock:
-            if self._ev is None:
-                # in the case where either have never been
-                # called or have already fully cycled once
-                if self._should_suspend(value):
-                    self._ev = asyncio.Event(loop=self._loop)
+            if self._should_suspend(value):
+                self._alarm = True
+                if self._ev is None and self.RE is not None:
+                    loop = self.RE.loop
+                    self._ev = asyncio.Event(loop=loop)
 
-                    self._loop.call_soon_threadsafe(
+                    loop.call_soon_threadsafe(
                         self.RE.request_suspend,
                         self._ev.wait())
-            else:
-                if self._should_resume(value):
+            elif self._should_resume(value):
+                self._alarm = False
+                if self._ev:
+
                     ev = self._ev.set
                     sleep = self._sleep
+                    if self.RE is not None:
+                        loop = self.RE.loop
+
+                        def local():
+                            loop.call_later(sleep, ev)
+                        loop.call_soon_threadsafe(local)
+                # clear that we have an event
+                self._ev = None
 
-                    def local():
-                        self._loop.call_later(sleep, ev)
-                    self._loop.call_soon_threadsafe(local)
-                    # clear that we have an event
-                    self._ev = None
+    @property
+    def alarm(self):
```

Comment:
As we discussed yesterday, unless you plan on following the minor/major/etc alarm schemes of EPICS, it may be better to rename these.